src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java

```diff
@@ -0,0 +1,125 @@
+
+    private final ItemRequestDtoMapper requestMapper;
+
+    private final ItemRequestRepository requestStorage;
+
+    private final UserRepository userStorage;
+
+    private final ItemRepository itemStorage;
+
+    @Transactional
+    @Override
+    public ItemRequestReplyDto addRequest(Long userId, ItemRequestDto dto) {
+        User requester = findRequester(userId);
+        ItemRequest request = requestMapper.fromDto(dto, requester, List.of(), LocalDateTime.now());
+        ItemRequest created = requestStorage.save(request);
+        log.info("New item-request added with new id {}", created.getId());
+        return requestMapper.toDto(created, List.of());
+    }
+
+    @Override
+    public List<ItemRequestReplyDto> getRequestsByUserId(Long userId) {
+        findRequester(userId);
+        List<ItemRequest> requests = requestStorage.findByRequesterIdOrderByCreatedAsc(userId);
+        Map<Long, Item> requestIdMapsItem = getItemsMadeForRequests(requests);
+        return mapItemRequestsToRequestsReplyDto(requests, requestIdMapsItem);
+    }
+
+    @Override
+    public List<ItemRequestReplyDto> getAllRequestsByAnotherUsers(Long from, Integer size, Long userId) {
+        List<ItemRequest> requests = requestStorage.findAllByRequesterIdNot(userId,
+                PageRequest.of((int) (from / size), size, Sort.by("created").ascending()));
+        Map<Long, Item> requestIdMapsItem = getItemsMadeForRequests(requests);
+        return mapItemRequestsToRequestsReplyDto(requests, requestIdMapsItem);
+    }
+
+    @Override
+    public ItemRequestReplyDto getRequestById(Long requestId, Long userId) {
+        findRequester(userId);
+        ItemRequest request = findRequest(requestId);
+        Map<Long, Item> requestIdMapsItem = getItemsMadeForRequests(List.of(request));
+        return requestMapper.toDto(request, new ArrayList<>(requestIdMapsItem.values()));
+    }
+
+    /**
+     * внутренний метод поиска пользователя по идентификатору
+     * @param userId идентификатор пользователя
+     * @return объект сущность User - пользователь, размещающий запрос на вещь
+     */
+    private User findRequester(Long userId) {
+        return userStorage.findById(userId).orElseThrow(
+                () -> {
+                    log.info("User with id {} not found", userId);
+                    throw new NotFoundException(format("user with id %d not found", userId));
+                }
+        );
+    }
+
+    /**
+     * вспомогательный метод запроса на вещь из БД
+     * @param requestId идентификатор запроса на вещь
+     * @return сущность вещи, если она найдена в базе
+     */
+    private ItemRequest findRequest(Long requestId) {
+        return requestStorage.findById(requestId).orElseThrow(
+                () -> {
+                    log.info("item-request with id {} not found", requestId);
+                    throw new NotFoundException(format("item-request with id %d not found", requestId));
```

Comment:
Так как здесь блок кода, а не просто лямбда выражение, лучше делать 
```suggestion
                    return new NotFoundException(format("item-request with id %d not found", requestId));
```
На уровне выше в `()->` перехватится возвращаемая ошибка и выбросится наружу

Reply:
если я правильно понял - это нужно сделать для всех orElseThrow в коде.. т.к. это будет соответствовать интерфейсу Supplier<Supplier<? extends X>, где X extends Throwable

Reply:
Только в местах, где еще логируешь что-то